### PR TITLE
Initial roi feature in segment app 

### PR DESCRIPTION
**Pull request by leoarc**
## Description

- Initial roi feature in the segment app .

- Feature to extract roi from selected region

Other small changes :

- Added progress percentage in snackbar

- Typo fix

- Use innerText instead of innerHTML

## How Has This Been Tested?
OS : Ubuntu 20.04
Browser : Firefox

## Types of changes
**What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
**Go over all the following points, and put an `x` in all the boxes that apply.
**(If you're unsure about any of these, don't hesitate to ask. We're here to help!)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by leoarc** [comment]
@birm why are the tests not been triggered ?

**Comment by birm** [comment]
Looks like it's here https://travis-ci.org/github/camicroscope/caMicroscope/builds/697929991
Not sure why it's not showing up.

File: apps/segment/segment.js [hunk 1]
```diff
@@ -1178,13 +1253,13 @@ async function showInfo() {
           store.get(name).onsuccess = function(e) {
             inputShape = e.target.result.input_shape.slice(1, 3).join('x');
             td = row.insertCell();
-            td.innerHTML = name.split('_').splice(1).join('_').slice(0, -3);
+            td.innerText = name.split('_').splice(1).join('_').slice(0, -3);
```
Comment (birm) [inline apps/segment/segment.js:1256]: Thanks for addressing these kinds of things!

**Comment by birm** [comment]
@Insiyaa I'm sorry I still can't add you as a reviewer directly.

**Comment by Insiyaa** [comment]
Sorry for the delay, I'll review as soon as possible

**Comment by leoarc** [comment]
@birm  @Insiyaa  I am adding to this since this is open . Hope that's ok ?

File: apps/model/model.js [hunk 2]
```diff
@@ -194,7 +195,7 @@ async function initUIcomponents() {
     provideContent: true,
     content: `
     <div class= "message" >
-      <h3> Select the parameters for the patcehs that you want to downlaod</h3></div><br>
+      <h3> Select the parameters for the patches that you want to downlaod</h3></div><br>
```
Comment (birm) [inline apps/model/model.js:198]:
```suggestion
      <h3> Select the parameters for the patches that you want to download</h3></div><br>
```

File: apps/segment/segment.js [hunk 3]
```diff
@@ -134,6 +136,75 @@ async function initUIcomponents() {
...
+
+  $UI.choiceModal = new ModalBox({
+    id: 'choice_panel',
+    hasHeader: true,
+    headerText: 'Select Parameters',
+    hasFooter: false,
+    provideContent: true,
+    content: `
+    <div class= "message" >
+      <h3> Select the parameters for the patches that you want to downlaod</h3></div><br>
```
Comment (birm) [inline apps/segment/segment.js:178]:
```suggestion
      <h3> Select the parameters for the patches that you want to download</h3></div><br>
```

**Comment by Insiyaa** [comment]
Looks good. Please check for spaces around the operators, rest it looks good.

**Comment by leoarc** [comment]
@birm can you tell how to solve the html injection issue ? The input is coming from the db and not the user , so as long as the input is sanitized earlier it should be fine right ? Also as I need the html tags I can't use something like innerText .

**Comment by leoarc** [comment]
@Insiyaa can you please elaborate on what you mean by spaces around operators ?

**Comment by birm** [comment]
@leoarc the simplest solution would probably be to use a sanitize function to replace "<" and ">" with safe characters. See https://github.com/camicroscope/caMicroscope/blob/cd64138b319c37dde23a15dc3a1933c4ac63872f/apps/table.js#L1

Another option is to create each html node as is done here https://github.com/camicroscope/caMicroscope/blob/cd64138b319c37dde23a15dc3a1933c4ac63872f/apps/table.js#L800

**Comment by leoarc** [comment]
Does this work ?
Screenshot from 2020-06-28 22-30-17


Screenshot from 2020-06-28 22-30-42


Screenshot from 2020-06-28 22-31-04

**Comment by leoarc** [comment]
I may have missed some. Will do another pass in the next PR .

**Comment by leoarc** [comment]
Thank you for the reviews ! And sorry for the constant bugging .

**Pull request merged**